src/cpp/shared/combat/combat.cpp

```diff
@@ -426,3 +429,22 @@ int combatManager::ValidSpellTarget(int spell, int hexIdx) {
     return 0;
   }
 }
+
+void combatManager::SetupCombat(int arg0, int arg1, hero *h1, armyGroup *a1, town *t, hero *h2, armyGroup *a2, int arg2, int arg3, int arg4) {
+    SetupCombat_orig(arg0, arg1, h1, a1, t, h2, a2, arg2, arg3, arg4);
+    ironfist_exp.combat.stack.abilityCounter.clear();
+    ironfist_exp.combat.stack.abilityNowAnimating.clear();
+}
+
+void combatManager::ResetRound() {
+    ResetRound_orig();
+    for(int i = 0; i < 2; i++) {
+        for(int j = 0; j < 21; j++) {
```

Comment:
I think you mean "j < MAX_STACKS" 

https://github.com/jkoppel/project-ironfist/blob/master/src/cpp/shared/combat/combat.h#L115

Oh crap. I guess MAX_STACKS is defined in IDA, but not in Ironfist. Please fix.